### PR TITLE
Create survey api and text questions api 

### PR DESCRIPTION
**Pull request by AashnaNarang**
Issues:
- This closes https://github.com/AashnaNarang/SYSC4806/issues/11 and https://github.com/AashnaNarang/SYSC4806/issues/5

What's in the PR
- POST, PATCH, DELETE apis for surveys
- POST, PATCH, DELETE apis for text questions
- integration tests for surveys and text questions apis 
- Cascade delete the survey questions if you delete a survey

### CODE REVIEW

File: app/controllers/api/v1/surveys_controller.rb [hunk 1]
```diff
@@ -33,6 +43,15 @@ def create
 
   # PATCH/PUT /surveys/1 or /surveys/1.json
   def update
+    @survey.update(survey_params)
+    if @survey.isLive
+      @survey.wentLiveAt = DateTime.now()
+    end
+    if @survey.save
```
Comment (AndrewHall797) [inline app/controllers/api/v1/surveys_controller.rb:50]: update already calls save, so you don't need to add this. https://apidock.com/rails/ActiveRecord/Base/update/class

File: app/controllers/api/v1/surveys_controller.rb [hunk 2]
```diff
@@ -33,6 +43,15 @@ def create
 
   # PATCH/PUT /surveys/1 or /surveys/1.json
   def update
+    @survey.update(survey_params)
```
Comment (AndrewHall797) [inline app/controllers/api/v1/surveys_controller.rb:46]: Are we going to allow live surveys to be updated?

File: app/controllers/api/v1/textquestion_controller.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    else
+      render json: @textquestion.errors
+    end
+  end
+
+  # PATCH/PUT /textquestion/1 or /textquestion/1.json
+  def update
+    @textquestion.update(question_params)
+
+    if @textquestion.save
```
Comment (AndrewHall797) [inline app/controllers/api/v1/textquestion_controller.rb:24]: Same thing as the other one, just need to move the checks around update.

File: db/migrate/20220305173455_change_went_live_at.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class ChangeWentLiveAt < ActiveRecord::Migration[7.0]
+  def change
+      change_column :surveys, :wentLiveAt, :datetime
```
Comment (AndrewHall797) [inline db/migrate/20220305173455_change_went_live_at.rb:3]: ah my bad, I used the wrong type of time eh

File: app/controllers/api/v1/surveys_controller.rb (hunk 2 above)
Comment (AashnaNarang) [reply app/controllers/api/v1/surveys_controller.rb:46]: good catch

**Comment by AndrewHall797** [comment]
LGTM! The only thing you might wanna do before you merge is to remove the old test files, not sure if you mean to keep them in the PR for a reason.

**Review by AndrewHall797 — APPROVED** [review]

**Pull request merged**